lib/utils/utils.go

```diff
@@ -664,6 +665,39 @@ func ByteCount(b int64) string {
 		float64(b)/float64(div), "kMGTPE"[exp])
 }
 
+// IteratePages will iterate through pages of a paging function and apply the given function to each element of the results. It will use 0
+// as the page size to use the default max size of the paging function.
+func IteratePages[T types.Resource](ctx context.Context, pageFn func(context.Context, int, string) ([]T, string, error), applyFn func(T) error) error {
```

Comment:
My 2c:

- `pageFn` can capture the page size itself, so I would drop it here
- `ctx` can be captured in a similar manner, so we can drop it as well
- applyFn is not necessary, we can return the resulting []T at the end
- Supplying an initial token is a useful input to have
- Drop the `types.Resource` constraint, it's not necessary and makes this impossible to use with some APIs (DeviceTrustService, for example)

Callers that are more complex can always hand-craft their own loop, although I think there's little reason to do, even with the new constaints.

Put together:

```go


func Iterate[T any](initialPageToken string, nextPage func(pageToken string) ([]T, string, error)) ([]T, error) {
	var resp []T
	pageToken := initialPageToken
	for {
		var page []T
		var err error
		page, pageToken, err = pageFn(ctx, pageSize, pageToken)
		if err != nil {
			return trace.Wrap(err)
		}

		resp = append(resp, page...)

		if pageToken == "" {
			break
		}
	}
	return resp, nil
}
```

Reply:
There is some value in maintaining the apply function. I'm currently adding in a short circuit mechanism, so I can save a pagination call/loop iteration if I maintain such a feature.

I can see the rest, though. I'll add these in.